### PR TITLE
is2142/type-to-viewers-map (⚠️ devops)

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

- Creates new table in database of services that consume filetypes (N:N) to replace the tmp hard-coded solution
- change entrypoints of ``openapi-viewer`` section in the webserver API 
     - NOTE A viewer resource (in future will be call a ``view-link``) should be thought as a tuple ``(filetype, service, input-port)``

### In coming PRs

- Move ``openapi-viewer`` section in webserver API -> api-server with token authentication 
- Move ``services/web/server/src/simcore_service_webserver/studies_dispatcher/_core.py`` logic to catalog service, i.e. access to table that maps services with filetypes (since in the future other metadata will be part of the mapping)

### ⚠️ devops

- Delete deprecated environment variables ``WEBSERVER_VIEWER_RAWGRAPH_VERSION`` and ``WEBSERVER_VIEWER_SIM4LIFE_VERSION``
- Inject data in ``services_consume_filetypes`` table in pg-db
- ``WEBSERVER_STUDIES_ACCESS_ENABLED=1`` to enable deploying views and studies

## Related issue/s

- #2142: Move filetype-to-viewers map to database (currently hard-coded)

#### Auto-testing

```cmd
make devenv
cd services/web/server
make install-dev
make test-dev-unit
```

#### Manual exploratory testing
- build and deploy
```cmd
cd osparc-simcore
make build-x
make up
```
- register a user
- inject data in the pg database 
```sql
INSERT INTO "services_consume_filetypes" ("service_key", "service_version", "service_display_name", "service_input_port", "filetype", "preference_order") VALUES
('simcore/services/dynamic/bio-formats-web',	'1.0.1',	'bio-formats',	'input_1',	'PNG',	0),
('simcore/services/dynamic/raw-graphs',	'2.11.1',	'RAWGraphs',	'input_1',	'CSV',	0),
('simcore/services/dynamic/bio-formats-web',	'1.0.1',	'bio-formats',	'input_1',	'JPEG',	0),
('simcore/services/dynamic/raw-graphs',	'2.11.1',	'RAWGraphs',	'input_1',	'TSV',	0),
('simcore/services/dynamic/raw-graphs',	'2.11.1',	'RAWGraphs',	'input_1',	'XLSX',	0),
('simcore/services/dynamic/raw-graphs',	'2.11.1',	'RAWGraphs',	'input_1',	'JSON',	0);
```
- GO to http://127.0.0.1:9081/dev/doc#/viewer
- Try queries
- CURL some of the view_url by adding ``file_size``,[``file_name``] and ``download_link`` as query parameters

## Checklist

- [x] Database migration script
- [x] Send new redirection URL to @ignapas 
- [x] Did you change any service's API? Then make sure to bundle document and upgrade version (``make openapi-specs``, ``git commit ...`` and then ``make version-*``)
- [x] Unit tests for the changes exist
- [x] Runs in the swarm
- [x] Documentation reflects the changes
- [x] New module? Add your github username to [.github/CODEOWNERS](.github/CODEOWNERS)

### CODE REVIEW

**Review by sanderegg — APPROVED** [review]
some minor comments. great work!

**Review by ignapas — APPROVED** [review]
I think the data model is not ideal for representing the complexity of this. The difference between extensions and file types, the 1..n relationship between file types and extensions, the n..n relationship between viewers and file types, etc... also the response payload will most of the times need to be post processed / converted to a table in order to be usable:
* file type to array of viewers
* viewer to array of filetypes
It is like you a are sending the intermediate table that represents the n to n relationship between the two main tables.

**Comment by pcrespov** [comment]
> I think the data model is not ideal for representing the complexity of this. The difference between extensions and file types, the 1..n relationship between file types and extensions, the n..n relationship between viewers and file types, etc... also the response payload will most of the times need to be post processed / converted to a table in order to be usable:
> 
> * file type to array of viewers
> * viewer to array of filetypes
> 
> It is like you are directly sending the intermediate table that represents the n to n relationship between the two main tables.

I agree, but given the use cases available I believe this implementation provides the maximum RoI.

File: services/web/server/src/simcore_service_webserver/studies_dispatcher/_core.py [hunk 1]
```diff
@@ -1,30 +1,47 @@
...
+    to visualize a file of that type
+    """
+
+    key: constr(regex=KEY_RE)
+    version: constr(regex=VERSION_RE)
+    filetype: str = Field(..., description="Filetype associated to this viewer")
+
+    label: str = Field(..., description="Display name")
+    input_port_key: str = Field(
+        "input_1",
```
[the remark this answers is not in the document]
Comment (pcrespov) [reply services/web/server/src/simcore_service_webserver/studies_dispatcher/_core.py]: agreed. I removed it and made it required from the db.

**Pull request merged**